extensions/amp-fx-flying-carpet/0.1/test/test-amp-fx-flying-carpet.js

```diff
@@ -167,18 +167,12 @@ describe('amp-fx-flying-carpet', () => {
       const posttext = iframe.doc.createTextNode('\n');
       return [pretext, img, posttext];
     }).then(flyingCarpet => {
-      sandbox.stub(
-        flyingCarpet.implementation_,
-        'attemptChangeHeight',
-        function(height, callback) {
-          flyingCarpet.style.height = height;
-          callback();
-        }
-      );
+      const attemptChangeHeight = sandbox.spy(flyingCarpet.implementation_,
+          'attemptChangeHeight');
       expect(flyingCarpet.getBoundingClientRect().height).to.be.gt(0);
       img.collapse();
-      expect(flyingCarpet.getBoundingClientRect().height).to.equal(0);
-      expect(flyingCarpet.style.display).to.equal('none');
+      expect(attemptChangeHeight).to.have.been.called;
```

Comment:
You are right, I could do that to test `collapse` gets called. But I think I would still not able test `expect(flyingCarpet.getBoundingClientRect().height).to.equal(0);`, since it would change the behavior of `attemptChangeHeight` function.